ur5_moveit_config/launch/moveit_rviz.launch

```diff
@@ -11,6 +12,7 @@
   <node name="$(anon rviz)" launch-prefix="$(arg launch_prefix)" pkg="rviz" type="rviz" respawn="false"
 	args="$(arg command_args)" output="screen">
     <rosparam command="load" file="$(find ur5_moveit_config)/config/kinematics.yaml"/>
+    <param name="arm_prefix" value="$(arg arm_prefix)"/>
```

Comment:
> The URKinematics plugin is loaded into RViz when the kinematics.yaml is parsed

I'm confused: afaik, kinematics plugins are loaded by MoveIt (from [moveit/Kinematics/C++ API](http://moveit.ros.org/wiki/Kinematics/C%2B%2B_API)):

> The `RobotModel` and `RobotState` classes are the core classes that give you access to the kinematics. [..]

The `kinematics.yaml` file is loaded onto the parameter server by roslaunch, and the parameters are then [used by MoveIt](http://moveit.ros.org/wiki/Kinematics/Configuration) to configure the referenced subclasses of `KinematicsBase`.

MoveIt also exposes the kinematics solvers in one of its [capabilities](http://moveit.ros.org/wiki/Capabilities#Kinematics).

RViz is not involved in this, other than that it provides an interface to most of the MoveGroup functionality (through the assorted services / action servers).

I don't currently know where that private `NodeHandle` in `URKinematicsPlugin` resolves to, but I don't really understand how it could be RViz?